### PR TITLE
Makes all Destroy() procs return properly

### LINKED ISSUE #10091: Nearly everything passed through qdel() does not have a QDEL_HINT attached to it.
Problem description: Almost every object that is passed through qdel falls into the 'else' category of this switch: https://github.com/tgstation/-tg-station/blob/master/code/controllers/subsystem/garbage.dm#L109-L123

This is due to most /Destroy() procs only containing a ..() at the end (ex: https://github.com/tgstation/-tg-station/blob/master/code/game/objects/objs.dm#L15) instead of a ". = ..()" or "return ..()"

What did you expect to happen: Every subtype of datum without a custom qdel hint to go into category 1, QDEL_HINT_QUEUE, as it is defined at the top in /datum/proc/Destroy()

What happened instead: Everything without a custom qdel hint skipped category 1 and ended up caught in the else.

Why is this bad/What are the consequences: There are no significant consequences on -tg-, as the garbage collector can handle datums and all subtypes. However, a potential issue is something being Queued that should definitely not be Queued. 

Steps to reproduce the problem: Add a "world << "gc switch skipped, [A], [A.type] went into else"" above [this line](https://github.com/tgstation/-tg-station/blob/master/code/controllers/subsystem/garbage.dm#L123). Observe everything without a custom qdel hint triggering the message.

Server: Local test server
Revision: 1f147b662e5ac0211fb7878926da6746b53456b5

### LINKED ISSUE #11251: Certain machines may be broken
If the machines destroy proc dosent have a ..() in it the machine may be used to make multiple copies of the machine parts when it is deconstructed.

### LINKED ISSUE #11367: Cyborgs are being qdelled instead of hard-deleted
This happens if there is still a variable that points to that Object. in order to soft delete properly, the Object's destroy proc needs to clear up all references that point to it.

Failed qdel log:
http://privatepaste.com/edc18b668d

### PR DESCRIPTION
**Pull request by duncathan**
Now every `Destroy()` will return either `..()` or its own qdel hint. This is useful because of my new `QDEL_HINT_FINDREFERENCE` from my last PR. Now if one is having a `qdel()` issue, they need only make the top-level type return `QDEL_HINT_FINDREFERENCE`, and all its children will do the same; after all, if the parent isn't being garbage collected gracefully, then the children sure as hell won't be. In addition, a `testing()` message (this type of message is displayed only when the `TESTING` compile option is enabled) will display a warning if an object's `Destroy()` proc is not returning a qdel hint.
This PR also makes a minor backend change to how `QDEL_HINT_FINDREFERENCE` and `qdel_then_find_references()` work.

Note that I assumed that anything not returning a value in `Destroy()` should return `..()`; if this is not the case with something please let me know.

Fixes #10091 
Fixes #11251
Fixes #11367

### CODE REVIEW

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -24,11 +24,6 @@
 	. = ..()
 	unbuckle_mob()
 
-/obj/Del()
-	. = ..()
-	unbuckle_mob()
-
```
Comment (duncathan) [inline code/game/objects/buckling.dm]: `Destroy()` is always called before any `Del()` call, so this was redundant.

File: code/game/objects/effects/effect_system/effects_smoke.dm [hunk 2]
```diff
@@ -192,8 +192,7 @@
 
 /datum/effect/effect/system/smoke_spread/chem/Destroy()
 	chemholder = null
-	..()
-	return QDEL_HINT_PUTINPOOL
```
Comment (duncathan) [inline code/game/objects/effects/effect_system/effects_smoke.dm]: the parent returns this hint, so it would now be redundant.

**Comment by Jalleo** [comment]
There may be a issue report still existing by me about a certain machine destroy that isnt fixed yet.

File: code/game/machinery/announcement_system.dm [hunk 3]
```diff
@@ -65,7 +65,7 @@ var/list/announcement_systems = list()
 
 /obj/machinery/announcement_system/Destroy()
 	announcement_systems -= src //"OH GOD WHY ARE THERE 100,000 LISTED ANNOUNCEMENT SYSTEMS?!!"
-	..()
+	return ..()
```
Comment (duncathan) [inline code/game/machinery/announcement_system.dm]: @Jalleo Is this what you're referring to?

Comment (Jalleo) [inline code/game/machinery/announcement_system.dm]: Yeah ya can say that issue report is fixed then

**Comment by duncathan** [comment]
>PR made 10 hours ago
>merge conflicts
>:'^)

**Comment by Fox-McCloud** [comment]
Now that this is being done, there's a few things that you may want to look into---namely that you're going to have some redundant Destroys---namely a lot of destroys set loc to null or do something that is handled by the parent proc---you'll want to go through and fix all of these up, eventually.

**Comment by duncathan** [comment]
@Fox-McCloud 
I will look into the redundant `Destroy()`s and `Del()`s in a later PR. As it is I want to get this one merged quick, as it'll greatly improve `qdel()` debugging.
As for `/mob/` returning `QDEL_HINT_QUEUE`; this PR has ensured that all `Destroy()` procs will return the same hint, `QDEL_HINT_QUEUE` being equivalent to no hint at all. If `/mob/` should not be returning that hint then that is outside the scope of this PR.

Also, conflicts have been fixed.

**Pull request merged**